src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Internal/HostTests.cs

```diff
@@ -1372,7 +1372,9 @@ private class AsyncThrowingService : BackgroundService
         {
             protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             {
-                await Task.Yield();
+                // Delay long enough that BackgroundService.StartAsync gets a chance to complete
+                // before the exception is thrown. But don't delay for too long so the test completes quickly.
+                await Task.Delay(50);
```

Comment:
Just from experience, I can almost guarantee this will still fail in CI at some point.

The goal here to ensure that this async method doesn't complete until after some predefined point in the test?  If so, can you instead structure this to await something the caller controls? e.g.
```C#
private class AsyncThrowingService : BackgroundService
{
    private readonly Task _t;

    public AsyncThrowingService(Task t) => _t = t;

    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
    {
        await _t;
        throw new Exception("Background Exception");
    }
}
```
and then in the test:
```C#
var tcs = new TaskCompletionSource<bool>();
var ats = new AsyncThrowingService(tcs.Task);
...
tcs.SetResult(true);
```
?

Reply:
Yeah, that's a much better idea.  Let me try that.